xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/XWikiServletURLFactory.java

```diff
@@ -444,23 +446,53 @@ public URL createExternalURL(String spaces, String name, String action, String q
         return this.createURL(spaces, name, action, querystring, anchor, xwikidb, context);
     }
 
+    private URL buildURL(URL serverUrl, String path, Map<String, Object> queryParameters, XWikiContext context)
+    {
+        try {
+            URIBuilder uriBuilder = new URIBuilder(serverUrl.toURI()).setPath(path);
+            for (Map.Entry<String, Object> queryParameter : queryParameters.entrySet()) {
+                String key = queryParameter.getKey();
+                Object paramValue = queryParameter.getValue();
+
+                if (paramValue instanceof String) {
+                    uriBuilder.addParameter(key, (String) paramValue);
+                } else if (paramValue instanceof String[]) {
```

Comment:
I would have supported any array but it's good enough with String[].

Reply:
> supported any array

Indeed. I'll improve it.